src/utils/url.js

```diff
@@ -156,3 +156,6 @@ export const autoCompleteUrl = (
   `${hasProtocol(value) ? '' : protocol}${value || 'your-org'}${
     value.indexOf('.') === -1 ? append : !value.match(/.+\..+\.+./g) ? shortAppend : ''
   }`;
+
+export const checkUrlValidity = (url: string) =>
+  /https:\/\/[a-zA-Z0-9-]+.[a-zA-Z0-9-]+.[a-zA-Z]+/.test(url) ? undefined : 'Invalid URL';
```

Comment:
While subdomains hosted on `zulipchat.com` might have this restriction on them that is unlikely the case for self-hosted apps that might have any domain.

Reply:
The fetch api on react-native android has this bug. So if we consider a case where special characters are allowed in the subdomain, the app will crash as soon as the fetch is called irrespective of using exception handling.

Reply:
This is potentially valid URL: 'hey.this.is.long.com' and it does not pass the Regex.